test/MOI/MOI_wrapper.jl

```diff
@@ -977,6 +988,260 @@ function test_primal_feasible_status()
+    f = 1.0 * x
+    MOI.set(model, MOI.ObjectiveFunction{typeof(f)}(), f)
+    g_bad = MOI.ScalarNonlinearFunction(:exp, Any[x])
+    c_bad = MOI.add_constraint(model, g_bad, MOI.GreaterThan(20.0))
+    g = MOI.ScalarNonlinearFunction(:*, Any[x, 2.0, x])
+    MOI.add_constraint(model, g, MOI.LessThan(3.0))
+    @test MOI.is_valid(model, c_bad)
+    MOI.delete(model, c_bad)
+    @test !MOI.is_valid(model, c_bad)
+    MOI.optimize!(model)
+    @test ≈(MOI.get(model, MOI.VariablePrimal(), x), sqrt(3 / 2); atol = 1e-3)
+    @test ≈(MOI.get(model, MOI.ObjectiveValue()), sqrt(3 / 2); atol = 1e-3)
+    return
+end
+
+function test_nonlinear_constraint_vector_delete()
+    if !Gurobi._supports_nonlinear()
+        return
+    end
+    model = Gurobi.Optimizer(GRB_ENV)
+    MOI.set(model, MOI.Silent(), true)
+    x = MOI.add_variable(model)
+    MOI.set(model, MOI.ObjectiveSense(), MOI.MAX_SENSE)
+    f = 1.0 * x
+    MOI.set(model, MOI.ObjectiveFunction{typeof(f)}(), f)
+    g_bad = MOI.ScalarNonlinearFunction(:exp, Any[x])
+    c_bad = MOI.add_constraint(model, g_bad, MOI.GreaterThan(20.0))
+    g = MOI.ScalarNonlinearFunction(:*, Any[x, 2.0, x])
+    MOI.add_constraint(model, g, MOI.LessThan(3.0))
+    @test MOI.is_valid(model, c_bad)
+    MOI.delete(model, [c_bad])
+    @test !MOI.is_valid(model, c_bad)
+    MOI.optimize!(model)
+    @test ≈(MOI.get(model, MOI.VariablePrimal(), x), sqrt(3 / 2); atol = 1e-3)
+    @test ≈(MOI.get(model, MOI.ObjectiveValue()), sqrt(3 / 2); atol = 1e-3)
+    return
+end
+
+function test_nonlinear_pow2()
+    if !Gurobi._supports_nonlinear()
+        return
+    end
+    model = Gurobi.Optimizer(GRB_ENV)
+    MOI.set(model, MOI.Silent(), true)
+    # max x + y
+    # s.t sqrt(^(x, 2) + ^(y, 2)) <= 1  # Use NL POW(2) operator
+    # x, y >= 0
+    #
+    # -> x = y = 1/sqrt(2)
+    x = MOI.add_variable(model)
+    y = MOI.add_variable(model)
+    MOI.set(model, MOI.ObjectiveSense(), MOI.MAX_SENSE)
+    f = 1.0 * x + 1.0 * y
+    MOI.set(model, MOI.ObjectiveFunction{typeof(f)}(), f)
+    f1 = MOI.ScalarNonlinearFunction(:^, Any[x, 2])
+    f2 = MOI.ScalarNonlinearFunction(:^, Any[y, 2])
+    f3 = MOI.ScalarNonlinearFunction(:+, Any[f1, f2])
+    g = MOI.ScalarNonlinearFunction(:sqrt, Any[f3])
+    c = MOI.add_constraint(model, g, MOI.LessThan(1.0))
+    MOI.optimize!(model)
+    @test ≈(MOI.get(model, MOI.ObjectiveValue()), 2 / sqrt(2); atol = 1e-3)
+    return
+end
+
+function test_nonlinear_scalarquadraticfunction()
+    if !Gurobi._supports_nonlinear()
+        return
+    end
+    model = Gurobi.Optimizer(GRB_ENV)
+    MOI.set(model, MOI.Silent(), true)
+    # max x + y
+    # s.t sqrt(*(x, x) + *(y, y)) <= 1
+    # x, y >= 0
+    #
+    # -> x = y = 1/sqrt(2)
+    x = MOI.add_variable(model)
+    y = MOI.add_variable(model)
+    MOI.set(model, MOI.ObjectiveSense(), MOI.MAX_SENSE)
+    f = 1.0 * x + 1.0 * y
+    MOI.set(model, MOI.ObjectiveFunction{typeof(f)}(), f)
+    f1 = MOI.ScalarNonlinearFunction(:*, Any[x, x])
```

Comment:
I made a few changes to the tests. In particular, this used to be `Any[x, y]`, but I think you intended `Any[x, x]`?